lib/src/binary_codec.dart

```diff
@@ -480,18 +503,23 @@ class PostgresBinaryDecoder<T> extends Converter<Uint8List?, T?> {
         return _decodeNumeric(input) as T;
 
       case PostgreSQLDataType.date:
-        return DateTime.utc(2000).add(Duration(days: buffer.getInt32(0))) as T;
-
+        try {
+          final value = buffer.getInt32(0);        
+          final date = DateTime.utc(2000).add(Duration(days: value));
+          return date as T;
+        } catch (e) {
```

Comment:
What exception is thrown here?

Reply:
PostgreSQL can have columns of type Timestamp defined with the value "infinity" which can generate an Exception, ideal for this case is to return null

Reply:
for example, when executing this query, it returns a column "rolvaliduntil" with the value 'infinity'

```sql
SELECT * FROM pg_authid where rolname ='postgres'
```